### PR TITLE
fix: link type detection for frameworks

### PR DESCRIPTION
**Pull request by cgrindel**
- The `find()` result for the file type results was broken. It was not detecting matches at the front of the string and it was not checking the dynamic result properly.
- Relaxed the static check to support additional file type results found in the wild.
- Add unit tests.

[Related Slack thread](https://bazelbuild.slack.com/archives/C05NZSHE7FU/p1698959858859589)

### CODE REVIEW

**Comment by cgrindel** [comment]
cc: @pswaminathan

File: swiftpkg/internal/artifact_infos.bzl [hunk 1]
```diff
@@ -89,18 +89,39 @@ def _new_framework_info_from_files(repository_ctx, path):
...
+    #   current ar archive
+    # dynamic Examples:
+    #   dynamically linked shared library
+    if file_type.find("ar archive") >= 0:
+        link_type = link_types.static
+    elif file_type.find("dynamic") >= 0:
+        link_type = link_types.dynamic
+    else:
+        link_type = link_types.unknown
+    return link_type
```
Comment (luispadron) [inline swiftpkg/internal/artifact_infos.bzl:123]: nit: could just return in each branch vs. setting `link_type`

**Review by luispadron — APPROVED** [review]

File: swiftpkg/internal/artifact_infos.bzl (hunk 1 above)
Comment (cgrindel) [reply swiftpkg/internal/artifact_infos.bzl]: Done.

**Comment by pswaminathan** [comment]
Our app passed ASC validation with this branch—thanks @cgrindel !

**Pull request merged**